src/index.js

```diff
@@ -12,6 +23,105 @@ export default class EtcdClient {
   }
 
   async start() {
-    return this.etcd;
+    return this;
+  }
+
+  finishCall(callInfo, status) {
+    this.emit('finish', { status, ...callInfo });
+  }
+
+  async get(context, key) {
+    const callInfo = {
+      client: this,
+      context,
+      key,
+      method: 'get',
+    };
+    this.emit('start', callInfo);
+    return new Promise((accept, reject) => {
+      this.etcd.get(key, (error, value) => {
+        this.finishCall(callInfo, statusCode(error));
+        if (error && error.errorCode === 100) {
+          accept();
+        } else if (error) {
+          reject(error);
+        }
+        accept(value ? value.node.value : null);
+      });
+    });
+  }
+
+  /**
+   * ttl is in seconds
+   */
+  async set(context, key, value, ttl) {
+    const callInfo = {
+      client: this,
+      context,
+      key,
+      value,
+      ttl,
+      method: 'set',
+    };
+    this.emit('start', callInfo);
+    return new Promise((accept, reject) => {
+      this.etcd.set(key, value, { ttl }, (error) => {
+        this.finishCall(callInfo, statusCode(error));
+        if (error) {
+          reject(error);
+        } else {
+          accept();
+        }
+      });
+    });
+  }
+
+  parseValue(v) {
+    if (v && v[0] === '[') {
+      try {
+        return JSON.parse(v);
+      } catch (error) {
+        // Do nothing, fall through
+      }
+    }
+    return [];
+  }
+
+  async acquireLock(context, key) {
+    const callInfo = {
+      client: this,
+      context,
+      key,
+      method: 'acquireLock',
+    };
+    this.emit('start', callInfo);
+    const lock = new Lock(this.etcd, key, os.hostname(), 10);
```

Comment:
I have an issue out with microlock about why this is a bad idea... So we should probably pick some other identifier, perhaps a passed in one. Otherwise multiple flows of execution on the same host can share the lock.

Reply:
What is the id supposed to represent? It looks like it's the name of the "node" in the docs? It seems like we need something else that is basically the host name or we should just use `key` for both parameters. Am I missing something?

Reply:
I see what you mean now. I'll use a uuid.